src/ParameterHandler.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Gergely Kovács
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "kroshu_ros2_core/ParameterHandler.hpp"
+
+#include <string>
+#include <vector>
+#include <memory>
+
+namespace kroshu_ros2_core
+{
+ParameterHandler::ParameterHandler(rclcpp_lifecycle::LifecycleNode * node)
+: node_(node)
+{
+}
+
+rcl_interfaces::msg::SetParametersResult ParameterHandler::onParamChange(
+  const std::vector<rclcpp::Parameter> & parameters) const
+{
+  rcl_interfaces::msg::SetParametersResult result;
+  result.successful = false;
+  for (const rclcpp::Parameter & param : parameters) {
+    auto found_param_it = std::find_if(
+      params_.begin(), params_.end(),
+      [&param](auto param_ptr) {
+        return param_ptr->getName() == param.get_name();
+      });
+    if (found_param_it == params_.end()) {
+      printf("Invalid parameter name\n");
```

Comment:
Add your remark as comment:

```suggestion
    // When used properly, we should not reach this, but better to keep additional check to filter improper use
    if (found_param_it == params_.end()) {
      printf("Invalid parameter name\n");
```